taichi/codegen/codegen_llvm.cpp

```diff
@@ -1201,14 +1228,44 @@ void CodeGenLLVM::visit(SNodeLookupStmt *stmt) {
 }
 
 void CodeGenLLVM::visit(GetChStmt *stmt) {
-  auto ch = create_call(
-      stmt->output_snode->get_ch_from_parent_func_name(),
-      {builder->CreateBitCast(llvm_val[stmt->input_ptr],
-                              llvm::PointerType::getInt8PtrTy(*llvm_context))});
-  llvm_val[stmt] = builder->CreateBitCast(
-      ch, llvm::PointerType::get(StructCompilerLLVM::get_llvm_node_type(
-                                     module.get(), stmt->output_snode),
-                                 0));
+  TI_INFO("GetChStmt input: {}", stmt->input_snode->get_node_type_name_hinted());
+  TI_INFO("GetChStmt ouptut: {}", stmt->output_snode->get_node_type_name_hinted());
+
+  if (stmt->output_snode->is_bit_level) {
+    // 1. create bit pointer struct
+    // struct bit_pointer {
+    //    i8* byte_ptr;
+    //    i32 offset;
+    // };
+    auto struct_type = llvm::StructType::get(*llvm_context, {llvm::Type::getInt8PtrTy(*llvm_context),
+                                                            llvm::Type::getInt32Ty(*llvm_context)});
+    // 2. alloca the bit pointer struct
+    auto bit_ptr_struct = create_entry_block_alloca(struct_type);
+
+    // 3. store the `input_ptr` in to `bit_struct`
+    auto byte_ptr = builder->CreateBitCast(llvm_val[stmt->input_ptr],
+                           llvm::PointerType::getInt8PtrTy(*llvm_context));
+
+    builder->CreateStore(byte_ptr, builder->CreateGEP(bit_ptr_struct,
+                                                      {tlctx->get_constant(0), tlctx->get_constant(0)}));
+    // 4. store `offset`
+    // temporarily use 0 for offset.
+    // TODO: correct offset.
+    builder->CreateStore(tlctx->get_constant(0), builder->CreateGEP(bit_ptr_struct,
+                                                                    {tlctx->get_constant(0), tlctx->get_constant(1)}));
```

Comment:
I am not sure how do we get the right offset here. So I use zero currently.

Reply:
`BitStructType::get_member_bit_offset` may be the right thing to look at. See also `GetChStmt::chid` :-)